lib/pbench/server/api/resources/__init__.py

```diff
@@ -1117,54 +1462,35 @@ def _dispatch(
                     HTTPStatus.INTERNAL_SERVER_ERROR,
                     message=HTTPStatus.INTERNAL_SERVER_ERROR.phrase,
                 )
-
-        try:
-            json_data = request.get_json()
-        except Exception as e:
-            self.logger.warning(
-                "{}: Bad JSON in request, {!r}, {!r}",
-                api_name,
-                str(e),
-                request.data,
-            )
-            abort(HTTPStatus.BAD_REQUEST, message="Invalid request payload")
-
-        try:
-            if json_data:
-                json_data.update(uri_parameters)
-            else:
-                json_data = uri_parameters
-            new_data = self.schema.validate(json_data)
-        except APIAbort as e:
-            self.logger.warning("{}: {} on {!r}", api_name, str(e), json_data)
-            abort(e.http_status, message=str(e))
-        except Exception as e:
-            self.logger.exception(
-                "{}: unexpected validation exception in {}: {}",
-                api_name,
-                e.__class__.__name__,
-                str(e),
-            )
-            abort(
-                HTTPStatus.INTERNAL_SERVER_ERROR,
-                message="INTERNAL ERROR IN VALIDATION",
+        else:
+            params = ApiParams(None, None, None)
+
+        # Automatically authorize the operation only if the API schema for the
+        # active HTTP method has authorization enabled, using the selected
+        # parameters. Automatic authorization can be disabled by selecting
+        # API_AUTHORIZATION.NONE for a particular method's schema where either
+        # no authorization is required, or a specialized authorization
+        # mechanism is required by the API.
+        auth_params = self.schemas.authorize(method, params)
+        if auth_params:
+            self.logger.info(
+                "Authorizing with {}:{}, {}",
+                auth_params.user,
+                auth_params.access,
+                auth_params.role,
             )
-
-        # Automatically authorize the operation only if the API schema has the
-        # "user" key of type USERNAME; otherwise we assume that authorization
-        # is unnecessary or that the API-specific subclass will take care of
-        # that in preprocess.
-        if "user" in self.schema and self.schema["user"].type is ParamType.USER:
-            user = new_data.get("user")  # converted user ID
-            access = new_data.get("access")  # normalized access policy
             try:
-                self._check_authorization(user, access)
+                self._check_authorization(
+                    auth_params.user, auth_params.access, auth_params.role
+                )
```

Comment:
Maybe; it evolved piecemeal. I'll have to check how/where it's called.

Reply:
`_check_authorization` is called in two places: one of them is PUT /datasets/metadata where the schema authorization model is `NONE` because the `user` namespace authorization is a bit different. So it's a toss-up. Yeah, there's now a convenient class to encapsulate the three, and I can just create an instance in the `_put`; but basically it just makes this call site shorter at the expense of making the other longer, while making the parameters a bit less obvious.